lib/_http_server.js

```diff
@@ -482,8 +483,15 @@ function connectionListener(socket) {
 
       res.detachSocket(socket);
 
-      if (res._last) {
-        socket.destroySoon();
+      if (res._last) socket.destroySoon();
+      else if (outgoing.length === 0) {
+        // When outgoing queue is empty we will wait for
+        // server.keepAliveTimeout and destroy socket not emiting
+        // 'timeout' events in req, res and server objects
+        socket.removeAllListeners('timeout');
+        socket.setTimeout(self.keepAliveTimeout, function onKeepAliveTimeout() {
```

Comment:
Another note, timeout still does not reset after receiving next request. Suppose you have following situation:

1. Request arrived
2. Response sent
3. `outgoing.length === 0`
4. `setTimeout(5s)`
5. 2 seconds passed
6. New request arrived
7. It is sending data very slowly

Now if the request will arrive after 2 seconds, the timeout value would be `5s` for it. While if it will have arrived immediately in a pipelined fashion - the timeout would be much bigger (the default value).

I think this PR should reset the timeout to a regular value upon receiving the pipelined request.